### ISSUE TITLE
Test failures in Github Actions for Python 3.10

### ISSUE THREAD

**Issue by tripleee** [Charcoal-SE/SmokeDetector#7100]
### What problem has occurred? What issues has it caused?

Many (but not all) test runs in the recent past has failed for 3.10 because the Python installed in .../SmokeDetector/Smokedetector/env/bin/python is a symlink to a version which does not exist.

In some more detail, test runs in my personal fork with `ls -l` interspersed before the failure reveal that .../env/bin/python3 (which python symlinks to) is a symlink to `/opt/hostedtoolcache/Python/3.10.4/x64/bin/python3` which however does not exist; `/opt/hostedtoolcache/Python` contains a directory for `3.10.5` (sic) but not `3.10.4`.

### What would you like to happen/not happen?

Ideally, I'm hoping this is a problem Github can solve without our involvement, but I'm creating a bug report for visibility and possible workarounds in the meantime.

**Comment by tripleee** [comment]
Actually I'm guessing the bug is in how we use `cache-runner` to restore `env` from possibly a different Python minor version.

**Comment by makyen** [comment]
The cache was cleared after GitHub's update from 3.10.4 to 3.10.5. In fact, that was the first thing I tried in order to resolve the issue. In order to do so, I changed the base storage/restore key, such that it wouldn't find the old caches. Given that we can't manually clear the caches for Github Actions, that's the primary way to invalidate the cache. In the past, I added a version number to the string, so that it would be easier to do so.

Hmmm.... if the cache contains old data, which it appears it might, then the problem could be that GitHub is jumping back and forth as to which version of Python is being used (e.g. some of their servers are using the old version and some the new version). I'll check for that.
